src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -77,14 +74,155 @@ public void registered(SchedulerDriver driver, FrameworkID frameworkId, MasterIn
+            reloadConfigsOnAllRunningTasks(driver);
+            sendMessageTo(
+                driver,
+                liveState.getFirstNameNodeTaskId(), liveState.getFirstNameNodeSlaveId(),
+                HDFSConstants.NAME_NODE_INIT_MESSAGE);
+            blockUntilNameNode1Healthy();
+            sendMessageTo(
+                driver,
+                liveState.getSecondNameNodeTaskId(), liveState.getSecondNameNodeSlaveId(),
+                HDFSConstants.NAME_NODE_BOOTSTRAP_MESSAGE);
+            liveState.transitionTo(AcquisitionPhase.DATA_NODES);
+          } else {
+            log.info("Cannot locate second namenode task id");
+          }
+          break;
+        case DATA_NODES :
+          break;
+      }
+    } else {
+      log.warn(String.format("Don't know how to handle state=%s for taskId=%s",
+          status.getState(), status.getTaskId().getValue()));
+    }
+  }
+
+  @Override
+  synchronized public void resourceOffers(SchedulerDriver driver, List<Offer> offers) {
+    log.info(String.format("Received %d offers", offers.size()));
+
+    if (liveState.getStagingTasksSize() != 0) {
+      log.info("Declining offers because tasks are currently staging");
+      for (Offer offer : offers) {
+        driver.declineOffer(offer.getId());
+      }
+    } else {
+      boolean acceptedOffer = false;
+      for (Offer offer : offers) {
+        if (acceptedOffer) {
+          driver.declineOffer(offer.getId());
+        } else {
+          switch (liveState.getCurrentAcquisitionPhase()) {
+            case JOURNAL_NODES :
+              if (liveState.getJournalNodeSize() < conf.getJournalNodeCount()) {
+                if (maybeLaunchJournalNode(driver, offer)) {
+                  acceptedOffer = true;
+                } else {
+                  driver.declineOffer(offer.getId());
+                }
+              }
+              break;
+            case NAME_NODE_1 :
+              if (maybeLaunchNameNode(driver, offer)) {
+                acceptedOffer = true;
+              } else {
+                driver.declineOffer(offer.getId());
+              }
+              break;
+            case NAME_NODE_2 :
+              if (maybeLaunchNameNode(driver, offer)) {
+                acceptedOffer = true;
+              } else {
+                driver.declineOffer(offer.getId());
+              }
+              break;
+            case DATA_NODES :
+              if (maybeLaunchDataNode(driver, offer)) {
+                acceptedOffer = true;
+              } else {
+                driver.declineOffer(offer.getId());
+              }
+              break;
+          }
+        }
+      }
+    }
+  }
+
+  @Override
+  public void slaveLost(SchedulerDriver driver, SlaveID slaveId) {
+    log.info("Slave lost slaveId=" + slaveId.getValue());
+  }
+
+  @Override
+  public void run() {
+    FrameworkInfo.Builder frameworkInfo = FrameworkInfo.newBuilder()
+        .setName("HDFS " + conf.getFrameworkName())
```

Comment:
We recently removed the "HDFS " part of the name, to comply with Mesos-DNS's requirement of no spaces in framework/task names. Please correct.